packages/gestalt/src/Masonry/defaultTwoColumnModuleLayout.js

```diff
@@ -278,14 +278,54 @@ const defaultTwoColumnModuleLayout = <T>({
     };
 
     if (hasTwoColumnItems) {
+      const prevOneColumnItems = [...itemsWithPositions];
+      let batchWithTwoColumnItem: $ReadOnlyArray<T> = [];
+
+      // If the number of items to position is greater that the batch size
+      // we identify the batch with the two column item and graph only that one
+      if (itemsWithoutPositions.length > TWO_COL_ITEMS_MEASURE_BATCH_SIZE) {
+        let currentBatch: Array<T> = [];
+
+        for (
+          let index = 0;
+          index < itemsWithoutPositions.length;
+          index += TWO_COL_ITEMS_MEASURE_BATCH_SIZE
+        ) {
+          currentBatch = itemsWithoutPositions.slice(
+            index,
+            index + TWO_COL_ITEMS_MEASURE_BATCH_SIZE,
+          );
+          // $FlowFixMe[incompatible-use] We're assuming `columnSpan` exists
+          if (currentBatch.some((item) => item.columnSpan > 1)) {
+            batchWithTwoColumnItem = [...currentBatch];
+          } else {
+            prevOneColumnItems.push(...currentBatch);
```

Comment:
this also has the effect of "pushing up" later one column items (again, very dependent on what the incoming items looks like)

assuming we have 24 items (so 4 batches) and the two column item is in the second batch, this would result in batches 1, 3 and 4 being positioned before the second batch.

instead i think what we want to do is split into three batches:
- all items before two column batch
- batch containing two column item
- everything else

first batch would then be `prevOneColumnItems`, second batch would be `batchWithTwoColumnItems` and third batch would be handled already at the end via https://github.com/pinterest/gestalt/pull/3458

Reply:
Changed to a more straightforward solution creating only three parts, `pre` before the two column batch, the two column batch and the rest.